Clean warning message

VitalyFedyunin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34165 Fix test_cuda
* #34164 Fix test/test_quantization.py
* #34163 Fix test_autograd
* #34156 Fix test_jit (partially)
* #34155 Fix test_nn and common_nn
* #34154 AssertEqual now checks tensors dtype
* **#34143 Clean warning message**

Differential Revision: [D20228174](https://our.internmc.facebook.com/intern/diff/D20228174)

VitalyFedyunin committed:
Clean warning message

[ghstack-poisoned]

gchanan commented on c10/core/TensorImpl.h:
```diff
@@ -1362,7 +1362,9 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
         TORCH_CHECK(false, "unsupported memory format ", memory_format);
         //TODO Implement set_sizes_and_strides for channels last 3d
-        break;
+        // Cleaning warning messages, no need to break as TORCH_CHECK(false)
```
couldn't you just remove these lines?

VitalyFedyunin replied:
Removing them is kind of error prone, if someone decide to lift "unsupported mem format" restriction.

VitalyFedyunin committed:
Update on "Clean warning message"


Differential Revision: [D20228174](https://our.internmc.facebook.com/intern/diff/D20228174)

[ghstack-poisoned]
